### ISSUE TITLE
Signals and Domains

### ISSUE THREAD

**Issue by kayceesrk** [ocaml-multicore/ocaml-multicore#334]
* Analyze the state of signal handling in Multicore OCaml
* Come up with an RFC for handling signals in multicore
  + As usual, don't break sequential behavior unless necessary.

**Comment by kayceesrk** [comment]
Relevant issues 
* EINTR-based signals: https://github.com/ocaml/ocaml/pull/1128

**Comment by ctk21** [comment]
Another relevant issue for signals and multiple domains of execution is:
- Fix Thread.sigmask: [ocam/ocaml/#2104](https://github.com/ocaml/ocaml/pull/2104) & [ocaml/ocaml/#2211](https://github.com/ocaml/ocaml/pull/2211)

**Comment by gadmm** [comment]
I would also like to mention the following race observed by Jacques-Henri and documented in `signals.c` since 4.10 (with the hope that the reworking of signals part of multicore might fix it):
```
   FIXME: We could get into caml_process_pending_actions when
   caml_something_to_do is seen as set but not caml_pending_signals,
   making us miss the signal.
```

**Comment by kayceesrk** [comment]
Fixed by #630, #631 and #649.
